tests/test_EclFile.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+   +   Copyright 2019 Equinor ASA.
+   +
+   +   This file is part of the Open Porous Media project (OPM).
+   +
+   +   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
+   +   the Free Software Foundation, either version 3 of the License, or
+   +   (at your option) any later version.
+   +
+   +   OPM is distributed in the hope that it will be useful,
+   +   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   +   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   +   GNU General Public License for more details.
+   +
+   +   You should have received a copy of the GNU General Public License
+   +   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+   +   */
+
+
+#include "config.h"
+#include <iostream>
+#include <stdio.h>
+#include <iomanip>    
+#include <math.h>  
+#include <algorithm>
+
+#include <examples/test_util/EclFile.hpp>
+
+#define BOOST_TEST_MODULE Test EclFile
+#include <boost/test/unit_test.hpp>
+
+
+
+BOOST_AUTO_TEST_CASE(TestConstructor){
+
+    EclFile file1("../../tests/NORNE_ATW2013.INIT");
```

Comment:
This use of path will unfortunately not work in general. The tests will be run with `cwd = ${CMAKE_BINARY_DIR}/tests` - and that is completely up to the user. A quite common configuration is to make a `build/` directory inside the root of `opm-common/` - and then this will work, but people/developers are free to make this directory anywhere - and then the test will not find the `NORNE_ATW2013.INIT` file. 

This is slightly awkward in my opinion - and if possible I strongly prefer to inline the test data; that is not an option here and then I think you should register the test file with the [here](https://github.com/OPM/opm-common/blob/master/CMakeLists_files.cmake#L287) - then the build system will copy the file to the `build/tests` directory and the code in the test can just be:
```
EclFile file1("NORNE_ATW2013.INIT");
```
when you are updating this any way you could choose a smaller file - e.g. the `SMSPEC` file from the SPE1 case.